### PR TITLE
Update from Intelsdi-x pluginsync 0.1.6

### PR DESCRIPTION
**Pull request by nanliu**

### CODE REVIEW

**Comment by nanliu** [comment]
facter is not on the package whitelist, so we may need to drop the medium tests.

**Comment by nanliu** [comment]
Note: Facter 1.x core dumps on travis.ci, and Facter 3.x is not compatible with this plugin.

**Comment by IRCody** [comment]
@nanliu: The medium tests seem to be working? I'm unsure what to expect given your comments. Can you clarify?

**Comment by nanliu** [comment]
@IRCody the old test was running with gem installed facter, the change to medium test allows it to work with packaged release of facter 3.x. There were significant restructure of facter output, and I had to trim quite a few assertions. The new facter output is visible in test here:

https://travis-ci.org/intelsdi-x/snap-plugin-collector-facter/jobs/162027506#L448-L767

Because of these changes I added another commit regarding the changes to test.

**Comment by IRCody** [comment]
So does the plugin support facter 2.x? Should calling out what versions are supported/tested be called out in the README?

**Comment by nanliu** [comment]
I made a note in the README regarding Facter versions.

File: README.md [hunk 1]
```diff
@@ -18,9 +18,11 @@ It's used in the [snap framework](http://github.com:intelsdi-x/snap).
 
 ## Getting Started
 ### System Requirements
-* [facter](https://github.com/puppetlabs/facter/blob/master/README.md) 
+* [facter 3.x](https://github.com/puppetlabs/facter/blob/master/README.md)
 * [golang 1.5+](https://golang.org/dl/) (needed only for building)
 
+NOTE: Facter 2.x may also work, but it's not currently not part of our test matrix.
```
Comment (IRCody) [inline README.md]: Remove one of the `not`s.

Comment (nanliu) [inline README.md]: Whoops, removed.

**Pull request merged**